src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Windows.cs

```diff
@@ -15,71 +15,197 @@ namespace System.Net.Http
 {
     // This implementation uses the System.Net.Http.WinHttpHandler class on Windows.  Other platforms will need to use
     // their own platform specific implementation.
-    public class HttpClientHandler : HttpMessageHandler
+    public partial class HttpClientHandler : HttpMessageHandler
     {
-        #region Properties
+        private readonly WinHttpHandler _winHttpHandler;
+        private readonly ManagedHandler _managedHandler;
+        private readonly DiagnosticsHandler _diagnosticsHandler;
+        private bool _useProxy;
+        private bool _disposed;
 
-        public virtual bool SupportsAutomaticDecompression
+        public HttpClientHandler()
         {
-            get { return true; }
+            if (UseManagedHandler)
+            {
+                _managedHandler = new ManagedHandler();
+                _diagnosticsHandler = new DiagnosticsHandler(_managedHandler);
+            }
+            else
+            {
+                _winHttpHandler = new WinHttpHandler();
+                _diagnosticsHandler = new DiagnosticsHandler(_winHttpHandler);
+
+                // Adjust defaults to match current .NET Desktop HttpClientHandler (based on HWR stack).
+                AllowAutoRedirect = true;
+                AutomaticDecompression = HttpHandlerDefaults.DefaultAutomaticDecompression;
+                UseProxy = true;
+                UseCookies = true;
+                CookieContainer = new CookieContainer();
+                _winHttpHandler.DefaultProxyCredentials = null;
+                _winHttpHandler.ServerCredentials = null;
+
+                // The existing .NET Desktop HttpClientHandler based on the HWR stack uses only WinINet registry
+                // settings for the proxy.  This also includes supporting the "Automatic Detect a proxy" using
+                // WPAD protocol and PAC file. So, for app-compat, we will do the same for the default proxy setting.
+                _winHttpHandler.WindowsProxyUsePolicy = WindowsProxyUsePolicy.UseWinInetProxy;
+                _winHttpHandler.Proxy = null;
+
+                // Since the granular WinHttpHandler timeout properties are not exposed via the HttpClientHandler API,
+                // we need to set them to infinite and allow the HttpClient.Timeout property to have precedence.
+                _winHttpHandler.ReceiveHeadersTimeout = Timeout.InfiniteTimeSpan;
+                _winHttpHandler.ReceiveDataTimeout = Timeout.InfiniteTimeSpan;
+                _winHttpHandler.SendTimeout = Timeout.InfiniteTimeSpan;
+            }
         }
 
-        public virtual bool SupportsProxy
+        protected override void Dispose(bool disposing)
         {
-            get { return true; }
-        }
+            if (disposing && !_disposed)
+            {
+                _disposed = true;
+                if (_winHttpHandler != null)
```

Comment:
The large number of these if/else pairs in this file suggests that you should be making a virtual call instead of switching between almost identical implementations